src/target/cortexm.c

```diff
@@ -442,7 +454,7 @@ bool cortexm_probe(ADIv5_AP_t *ap)
 				PROBE(rp_probe);
 			PROBE(lpc11xx_probe); /* LPC8 */
 		} else if (ap->ap_partno == 0x4c3)  { /* Cortex-M3 ROM */
-			PROBE(stm32f1_probe); /* Care for STM32F1 clones */
+			PROBE(stm32f1_clones_probe); /* Care for STM32F1 clones */
```

Comment:
This is an interesting approach and we can see why you've done it this way, but I think it'd be better to stick the CH32F1 probe routine here as another normal PROBE() call and treat it as its own target rather than a weird subtype of STM32F1